File: EntityModels.Postgresql/Region.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EntityModels.Postgresql;

[Table("Region")]
public partial class Region
{
    [Key] public string Name { get; set; } = null!;
    [StringLength(1000)] public string Description { get; set; } = null!;
    public int Order { get; set; }

    public virtual ICollection<Territory> Territories { get; set; } = null!;
}

File: client/src/models/region/Region.ts
import { Territory } from "./Territory";

export class Region {
  constructor(
    name: string,
    order: number,
    description: string,
    territories: Territory[]) {
    this.name = name;
    this.order = order;
    this.description = description;
    this.territories = territories;
  }

  name: string;
  order: number;
  description: string;
  territories: Territory[];
}